### ISSUE TITLE
Running Packer inside Docker container kills the agent container on commit/export

### ISSUE THREAD

**Issue by anupclouddevops** [hashicorp/packer#8597]
When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

I am running packer inside Jenkins build agents which are themselves Docker containers on AWS ECS and then I am using Packer docker builder to build our application docker image.
All provisioners run successfully and the image gets committed / exported perfectly.
However, when packer kills the docker container (inside container which it provisioned) , it also kills the Jenkins slave agent container and hence the build fails and post-processor tasks are not executed.

#### Reproduction Steps

Running Packer inside a container and using Packer docker builder

### Packer version

From 1.4.2

### Simplified Packer Buildfile

Can be provided if needed

### Operating system and Environment details

CentOS and jenkins agents running on AWS ECS (docker containers built on CentOS)

### Log Fragments and crash.log files

None

Set the env var `PACKER_LOG=1` for maximum log detail.

**Comment by nywilken** [comment]
Thanks for reaching out. In order for us to best help could you provide the things requested in the issue template; including the simplest Packer configuration and scripts needed to reproduce the bug.

> it also kills the Jenkins slave agent container and hence the build fails and post-processor tasks are not executed

When the Jenkins slave agent container is killed is there a process running within the Container? Is ECS killing the container? The Packer log information may help provide insight here.

**Comment by nywilken** [comment]
Hi @anupclouddevops are you still running into issues here? If so, can you provide a simplified version of your build configuration as part of your reproduction steps. Thanks!

**Comment by anupclouddevops** [comment]
I have found the issue where I was use PACKER_TMP_DIR = env. WORKSPACE which Packer was mounting to the container to transfer files. When Packer killed the temporary container, this volume was wiped and the workspace which had the artifact/manifest file was also getting deleted. This resulted in the failure of post processors and hence Jenkins marked the build as failure. The ECS plugin then stopped the task.

I have removed the above variable to let packer create a temp directory and its working fine now. 

Anyhow with Jenkins per build container, I don't have multiple docker builds running on the same instance and hence no need to specify a temp dir.

Thanka for your help.
